ironfish/src/network/peerNetwork.ts

```diff
@@ -998,34 +998,33 @@ export class PeerNetwork {
     const received = new Date()
 
     // Mark the peer as knowing about the transaction
-    const hash = new Transaction(message.transaction).hash()
+    const transaction = new Transaction(message.transaction)
+    const hash = transaction.hash()
     peer.state.identity && this.markKnowsTransaction(hash, peer.state.identity)
 
     // Let the fetcher know that a transaction was received and we no longer have to query it
     this.transactionFetcher.receivedTransaction(hash)
 
     if (this.shouldProcessTransactions() && !this.alreadyHaveTransaction(hash)) {
-      // Force lazy deserialization of the transaction as a first sanity check
-      const transaction = this.chain.verifier.verifyNewTransaction(message.transaction)
+      // Check that the transaction is valid
+      const { valid, reason } = await this.chain.verifier.verifyNewTransaction(transaction)
 
-      // Validate the transaction, so that the account and mempool do not receive
-      // an invalid transaction, and we do not gossip.
-      const { valid, reason } = await this.chain.verifier.verifyTransactionNoncontextual(
-        transaction,
-      )
       if (!valid) {
         Assert.isNotUndefined(reason)
-        this.logger.debug(
-          `Invalid transaction '${transaction.unsignedHash().toString('hex')}': ${reason}`,
-        )
+        // Logging hash because unsignedHash is slow
+        this.logger.debug(`Invalid transaction '${hash.toString('hex')}': ${reason}`)
+        // TODO: It might be more accurate to have a separate map of invalid transactions,
+        // but it accomplishes the same outcome of discarding the transaction in the future.
+        this.recentlyAddedToChain.set(hash, true)
```

Comment:
I think we should probably remove this just to keep the LRU consistent with transactions that are in the chain and not generally invalid transactions